dance/transforms/filter.py

```diff
@@ -268,6 +271,31 @@ def __init__(
                          inplace=inplace, **kwargs)
 
 
+@register_preprocessor("filter", "cell")
```

Comment:
FilterCellsCommonMod: Filters single-cell data to retain only common cells between two modalities.

This transformation filters two single-cell modalities (`mod1` and `mod2`) to retain only the cells
that are shared between them. Optionally, a solution dataset (`sol`) containing cell labels or annotations
can be filtered to match the common cells.